drivers/counter/counter_tmr_cmsdk_apb.c

```diff
@@ -63,18 +63,45 @@ static u32_t counter_tmr_cmsdk_apb_read(struct device *dev)
 	return value;
 }
 
-static int counter_tmr_cmsdk_apb_set_alarm(struct device *dev,
-					   counter_callback_t callback,
-					   u32_t count, void *user_data)
+static int counter_tmr_cmsdk_apb_set_top_value(struct device *dev,
+						u32_t ticks,
+						counter_top_callback_t callback,
+						void *user_data)
 {
 	return -ENODEV;
 }
 
+static u32_t counter_tmr_cmsdk_apb_get_top_value(struct device *dev)
+{
+	return TIMER_MAX_RELOAD;
+}
+
+static int counter_tmr_cmsdk_apb_set_alarm(struct device *dev, u8_t chan_id,
+				const struct counter_alarm_cfg *alarm_cfg)
+{
+	ARG_UNUSED(dev);
+	ARG_UNUSED(chan_id);
+	ARG_UNUSED(alarm_cfg);
+
+	return -ENODEV;
+}
+
+static int counter_tmr_cmsdk_apb_disable_alarm(struct device *dev, u8_t chan_id)
```

Comment:
Driver should provide info structure on driver initialization (https://github.com/zephyrproject-rtos/zephyr/blob/4d8f735fc635d51491d71a489c65c3b94d67a209/include/counter.h#L84). Info structure contains information about number of channels. If channels = 0 then `set_channel_alarm` and `disable_channel_alarm` might not be provided.

Reply:
note that counter api expects that info structure is set as it calls it unconditionally: https://github.com/zephyrproject-rtos/zephyr/blob/4d8f735fc635d51491d71a489c65c3b94d67a209/include/counter.h#L141